pennylane_qiskit/qiskit_device_legacy.py

```diff
@@ -342,9 +340,15 @@ def compile(self):
         If compile_backend is None, then the target is simply the
         backend.
         """
-        compile_backend = self.compile_backend or self.backend
-        compiled_circuits = transpile(self._circuit, backend=compile_backend, **self.transpile_args)
-        return compiled_circuits
+        return transpile(
+            self._circuit,
+            backend=self.compile_backend or self.backend,
+            **self.transpile_args,
+            # Disable routing to keep SWAP in circuit rather than
+            # using ElidePermutation to shuffle qubit indices around.
+            # See https://github.com/Qiskit/qiskit/issues/13144 for more details.
+            routing_method="none",
```

Comment:
I can see how this enables SWAP, but will it make it hard to run on hardware that needs to be transpiled with respect for physical routing constraints? I'm just wondering if maybe we're fixing one problem by creating a worse problem.

Reply:
You can use a fake backend with routing requirements. `GenericBackendV2` with a coupling map. I think we use it in some of the tests, and it shouldn't require hardware access.

Reply:
Discussed offline. This is only affecting the legacy device, which now is only used by simulators, so we don't need to worry about this after all 😌